modules/cudf/src/column/concatenate.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cudf/concatenate.hpp>
+#include <node_cudf/column.hpp>
+
+namespace nv {
+
+Column::wrapper_t Column::concat(cudf::column_view const& other,
+                                 rmm::mr::device_memory_resource* mr) {
+  try {
+    return Column::New(Env(), cudf::concatenate(std::vector{this->view(), other}, mr));
+  } catch (std::exception const& e) { NAPI_THROW(Napi::Error::New(Env(), e.what())); }
+}
+
+Napi::Value Column::concat(Napi::CallbackInfo const& info) {
+  CallbackArgs args{info};
+  try {
+    return concat(*Column::Unwrap(args[0]), args[1]);
+  } catch (cudf::logic_error const& e) { NAPI_THROW(Napi::Error::New(info.Env(), e.what())); }
+  throw Napi::Error::New(info.Env(), "Concat requires a Column");
```

Comment:
This logic seems a little off, since public `concat` already catches all exceptions. Even if it didn't, how would the last line be reach (if there were still some other uncaught exception after the `catch` it would just keep propagating up the stack, no?)

Recent pattern seems more along these lines:

* catch `catch (cudf::logic_error const& e)` in the public `concat` that actually calls `cudf` functions
* don't wrap the call to public `concat` in any `try...except`
* instead, do explicit argument number/type checking to raise "Concat requires a Column" before calling public `concat` if necessary

cc @trxcllnt

Reply:
Yup, the throw statement is completely useless in this case (for whatever reason assumed that the throw statement was apart of the `try...catch` block above).

and I'm pretty sure based off of this method definition
https://github.com/rapidsai/node-rapids/blob/ebc27db6c44cfa16d07e810c3712d7a679bf6ab2/modules/cudf/src/series.ts#L438

It shouldn't be possible to call it without a `Column`, so I don't think I even need any argument checking. (let me know if this isn't the case).

Thanks for the review.

Reply:
https://github.com/rapidsai/node-rapids/pull/183#discussion_r629810222

Only using `try...catch` inside of the `libcudf` call, removing other instance.